guardian/support-frontend
# Add subs payment method switches and consolidate payment method selector components

## What are you doing in this PR?
This PR adds debit, debit/credit card and paypal feature switches for subscriptions, which will allow us to switch off those payment methods in future if need be. 

The `countryPaymentMethods.ts` file which previously determined whether or not we would show the direct debit payment method option based on the country has been updated to account for whether subs payment method switches are turned on. This file has been renamed to the more generic `supportedPaymentMethods.ts` and the corresponding JSON files for each code environment have been updated in S3.

This also removes the contributions specific `PaymentMethodSelector.tsx` in favour of the subscriptions `paymentMethodSelector.tsx` component in an effort to consolidate more of our codebase and the related subscriptions checkout tests have been updated to account for different potential switch states.

When there is only 1 payment method available on subscriptions checkout the title will read "Payment Method" as opposed to "How would you like to pay?" and the only available payment method will be preselected:

<img width="498" alt="Screenshot 2022-08-01 at 03 37 43" src="https://user-images.githubusercontent.com/44685872/182389737-40126597-11a3-4cf2-b464-759728cd137d.png">

[**Trello Card**](https://trello.com/c/6JPATMMG/495-put-feature-switches-around-remaining-subscription-payment-methods-credit-card-paypal-client-side)

## How to test (locally)
- Pull this branch and start the support site locally
- Navigate to a subscriptions checkout page
- Start the support-admin-console locally
- Turn off a subs payment method switch in the RRCP
- Refresh the subs checkout page
- The payment method you switched off should no longer be available/visible on the subs checkout page
